Add a “dataList” field for deposits, orders, withdrawals, shifts etc

dmax10 opened the pull request.

dmax10 committed:
Refactor ExchangeRouter to reduce it's size

dmax10 committed:
Add support for bytes type in DataStore (bytesValues, getBytes, setBytes, removeBytes)

dmax10 committed:
WIP add data field for deposits, orders, withdrawals, shifts etc

dmax10 committed:
Merge branch 'v2.2-branch' into add-data-field

dmax10 committed:
Revert changes for the DataStore

dmax10 committed:
Replace bytes data field with bytes32[] dataList

dmax10 committed:
Fix tests: default dataList to empty array

dmax10 committed:
Fix tests: increase expectedPropsLength by one for the structs where the dataList field was added

dmax10 committed:
Fix "get, set, remove" tests: add _dataList to sampleItem

dmax10 committed:
Add data length validation

dmax10 committed:
Add dataList to Orders

dmax10 committed:
Revert "Refactor ExchangeRouter to reduce it's size"

This reverts commit 2bc7efab8d0f6046c0f55050a89e7b952200bb41.

dmax10 committed:
Merge branch 'refactor-config-contract-to-reduce-size' into add-data-field

dmax10 committed:
Add MAX_DATA_LENGTH to allowedBaseKeys

Keys which don't have a config setter are whitelisted as allowed and set dynamically

dmax10 committed:
Merge branch 'v2.2-branch' into add-data-field

dmax10 committed:
Comments update

dmax10 committed:
Revert dataList for GlvShift

GlvShift doesn't need dataList since createGlvShift is only used internally by keepers, so integrations can't use it

xvi10 commented on contracts/adl/AdlUtils.sol:
```diff
@@ -57,6 +57,7 @@ library AdlUtils {
         bool isLong;
         uint256 sizeDeltaUsd;
         uint256 updatedAtTime;
+        bytes32[] dataList;
```
should not be needed

dmax10 replied:
Removed

xvi10 commented on contracts/exchange/BaseHandler.sol:
```diff
@@ -50,4 +50,11 @@ contract BaseHandler is RoleModule, GlobalReentrancyGuard, OracleModule {
         }
     }
+
+    function validateDataSize(uint256 dataLength) internal view {
```
rename to validateDataListLength, doesn't seem to be used, should use this function

dmax10 replied:
Renamed and added validations for deposits, glvDeposits, withdrawals, shifts, orders

xvi10 commented on contracts/exchange/BaseHandler.sol:
```diff
@@ -50,4 +50,11 @@ contract BaseHandler is RoleModule, GlobalReentrancyGuard, OracleModule {
+    function validateDataSize(uint256 dataLength) internal view {
+        uint256 maxDataLength = dataStore.getUint(Keys.MAX_DATA_LENGTH);
+        if (dataLength > maxDataLength) {
+            revert Errors.MaxDataLengthExceeded(dataLength, maxDataLength);
```
rename to MaxDataListLengthExceeded

dmax10 replied:
Renamed

xvi10 commented on contracts/glv/glvDeposit/GlvDepositUtils.sol:
```diff
@@ -407,7 +409,8 @@ library GlvDepositUtils {
             }),
-            Deposit.Flags({shouldUnwrapNativeToken: false})
+            Deposit.Flags({shouldUnwrapNativeToken: false}),
+            glvDeposit.dataList()
```
dataList is to pass to callbacks, in here, the callbackContract is address(0), so there is no reason to pass glvDeposit.dataList(), should pass an empty list instead

dmax10 replied:
Changing it to `new bytes32[](0)` slightly  increases the bytecode (by 0.055 KiB) and throws with the contract size exceeded error. 

This would get solved though once the claimable-collateral-reduction-factor branch is merged because it includes some additional ExchangeRouter refactoring. 

Would update this suggestion once that branch is merged.

dmax10 replied:
`glvDeposit.dataList()` has been changed to `new bytes32[](0) // dataList`

xvi10 commented on utils/storeUtils.ts:
```diff
@@ -157,6 +157,7 @@ export async function validateStoreUtils({
       addresses: {},
       numbers: {},
       flags: {},
+      _dataList: [],
```
should pass in a non-empty list and add validations below that the dataList is properly set, updated and removed

dmax10 replied:
Added setSampleItemBytes32Array function down bellow which is dynamically setting values for dataList (similar to setting addresses / numbers / flags)

dmax10 committed:
Remove dataList from ADL orders

xvi10 commented on utils/deposit.ts:
```diff
@@ -71,6 +71,7 @@ export async function createDeposit(fixture, overrides: any = {}) {
     shouldUnwrapNativeToken,
     executionFee,
     callbackGasLimit,
+    dataList: [],
```
should allow dataList to be optionally passed in for deposits, withdrawals, etc

for the existing createDeposit, createWithdrawal, etc, tests, pass in a non-empty dataList and validate that it is properly set after create

dmax10 replied:
Made the dataList optional and tested with non-empty values for create*

dmax10 committed:
Rename error and fn for validating the dataList

dmax10 committed:
Allow dataList to be optionally passed in for deposits, shifts, withdrawals, glvDeposits, glvWithdrawals

dmax10 committed:
Set dataList field for CreateOrderParams

dmax10 committed:
Test dataList filed with non-empty values for deposits, shifts, withdrawals, glvDeposits, orders

dmax10 committed:
Add setSampleItemBytes32Array function which is dynamically setting values for dataList (similar to setting addresses / numbers / flags)

dmax10 committed:
Add dataList length validation for deposits, glvDeposits, withdrawals, shifts, orders

dmax10 committed:
Test dataList length validation for deposits, glvDeposits, withdrawals, shifts, orders

dmax10 committed:
Merge branch 'v2.2-branch' into add-data-field

dmax10 committed:
Refactor GlvDepositUtils to reduce bytecode size

Move _validateFirstGlvDeposit and _getMintAmount internal methods from GlvDepositUtils to the newly created GlvDepositHelper library.

dmax10 committed:
Use empty list for the dataList field since the callbackContract is address(0)

dmax10 committed:
Rename glv deposit utility lib

dmax10 committed:
Merge pull request #106 from gmx-io/refactor-glv-deposit-utils

Refactor glv deposit utils to reduce bytecode size

dmax10 committed:
Rename deploy script
